verilog/tools/formatter/verilog_format.cc

```diff
@@ -216,22 +194,62 @@ int main(int argc, char** argv) {
     }
     if (absl::GetFlag(FLAGS_failsafe_success)) {
       // original text was preserved, and --inplace modification is skipped.
-      return 0;
+      return true;
     }
-    return 1;
+    return false;
   }
 
   // Safe to write out result, having passed above verification.
   if (inplace && !is_stdin) {
-    if (!verible::file::SetContents(filename, formatted_output)) {
-      std::cerr << "Error writing to file: " << filename << std::endl;
-      std::cerr << "Printing to stdout instead." << std::endl;
-      std::cout << formatted_output;
-      return 1;
+    // Don't write if the output is exactly as the input, so that we don't mess
+    // with tools that look for timestamp changes (such as make).
+    if (content != formatted_output) {
+      if (!verible::file::SetContents(filename, formatted_output)) {
+        std::cerr << "Error writing to file: " << filename << std::endl;
+        std::cerr << "Printing to stdout instead." << std::endl;
+        std::cout << formatted_output;
+        return false;
+      }
     }
   } else {
     std::cout << formatted_output;
   }
 
-  return 0;
+  return true;
+}
+
+int main(int argc, char** argv) {
+  const auto usage = absl::StrCat("usage: ", argv[0],
+                                  " [options] <file>\n"
+                                  "To pipe from stdin, use '-' as <file>.");
```

Comment:
Should we now report a user-error if `-` is one of many files?  Few utilities have a meaningful behavior when stdin is one of multiple inputs, `diff` being one of them.

Reply:
Having stdin as one of many files is a benign situation (this is the only file we then output to stdout, which is exactly as expected).

But having multiple files without `--inplace` doesn't really make sense, so error on that now.